pkg/provider/beyondtrust/propvider_test.go

```diff
@@ -0,0 +1,231 @@
+	t.Run("should create new client", func(t *testing.T) {
+		ctx := context.Background()
+		p := &Provider{}
+		c := createMockPasswordSafeClient(t)
+		s := esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{
+					Beyondtrust: &esv1beta1.BeyondtrustProvider{
+						APIURL:        "https://example.com:443/BeyondTrust/api/public/v3/",
+						Retrievaltype: "SECRET",
+						Clientid: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "12345678-25fg-4b05-9ced-35e7dd5093ae",
+						},
+						Clientsecret: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "8i7U0Yulabon8mTc",
+						},
+					},
+				},
+			},
+		}
+
+		_, err := p.NewClient(ctx, &s, c, "test")
+
+		assert.Equal(t, err.Error(), "error: Key: 'UserInputValidaton.ClientSecret' Error:Field validation for 'ClientSecret' failed on the 'min' tag", "The two error should be the same.")
+	})
+}
+
+func TestNewClientBadSeparator(t *testing.T) {
+	t.Run("should create new client", func(t *testing.T) {
+		ctx := context.Background()
+		p := &Provider{}
+		c := createMockPasswordSafeClient(t)
+		s := esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{
+					Beyondtrust: &esv1beta1.BeyondtrustProvider{
+						APIURL:        "https://example.com:443/BeyondTrust/api/public/v3/",
+						Separator:     "//",
+						Retrievaltype: "SECRET",
+						Clientid: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "12345678-25fg-4b05-9ced-35e7dd5093ae",
+						},
+						Clientsecret: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "1234567890on8mTcOzJcltiEg4wYOhDVMer123456789",
+						},
+					},
+				},
+			},
+		}
+
+		_, err := p.NewClient(ctx, &s, c, "test")
+
+		assert.Equal(t, err.Error(), "error: Key: 'UserInputValidaton.Separator' Error:Field validation for 'Separator' failed on the 'max' tag")
+	})
+}
+
+func TestNewClientBadClientTimeOutinSeconds(t *testing.T) {
+	t.Run("should create new client", func(t *testing.T) {
+		ctx := context.Background()
+		p := &Provider{}
+		c := createMockPasswordSafeClient(t)
+		s := esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{
+					Beyondtrust: &esv1beta1.BeyondtrustProvider{
+						APIURL:               "https://example.com:443/BeyondTrust/api/public/v3/",
+						Separator:            "/",
+						Clienttimeoutseconds: 400,
+						Retrievaltype:        "SECRET",
+						Clientid: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "12345678-25fg-4b05-9ced-35e7dd5093ae",
+						},
+						Clientsecret: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "1234567890on8mTcOzJcltiEg4wYOhDVMer123456789",
+						},
+					},
+				},
+			},
+		}
+
+		_, err := p.NewClient(ctx, &s, c, "test")
+
+		assert.Equal(t, err.Error(), "error: Key: 'UserInputValidaton.ClientTimeOutinSeconds' Error:Field validation for 'ClientTimeOutinSeconds' failed on the 'lte' tag")
+	})
+}
```

Comment:
our approach is to use table driven tests. Can you make sure your tests follow that same approach for the sake of standardization?

Reply:
Yes, I took as reference others existing provider.

Reply:
This isn't table driven. :) It's using a t.Run inside the test which makes no sense. The table driven tests run the same thing in a loop but with different data supplied. Check out the other providers.

Reply:
Yes, you are totally right!, I've applied table driven, please check again, thank you.